analytics_data_api/v0/connections.py

```diff
@@ -0,0 +1,69 @@
+import json
+import time
+
+from boto.connection import AWSAuthConnection
+from django.conf import settings
+from elasticsearch import Connection
+
+
+class BotoHttpConnection(Connection):
+    """
+    Uses AWS configured connection to sign requests before they're sent to
+    elasticsearch nodes.
+    """
+
+    connection = None
+
+    def __init__(self, host='localhost', port=443, **kwargs):
+        super(BotoHttpConnection, self).__init__(host=host, port=port, **kwargs)
+        connection_params = {'host': host, 'port': port}
+
+        # Set the credentials through settings.  If not provided, boto will attempt to use
+        # default environment variables.
```

Comment:
We should consider accepting these variables as parameters to the constructor instead of reading them directly out of settings.

My pipeline code will not be able to import django.conf.settings.